### ISSUE TITLE
Is this actually detectable?

### ISSUE THREAD

**Issue by rawrchad** [iseekwonderful/csgoGlow#49]
Is it detectable through vac is what I am curious? and will it ever be you think if not already.

**Comment by sulik12345** [comment]
Vac does not work against MacOSX.

**Comment by rawrchad** [comment]
Why is that exactly? I would just like to understand.

**Comment by sulik12345** [comment]
VAC is a system that detects things running in your RAM and your game, so windows users using injectors to inject into different files would be VAC banned. For MAC however, the system is different, the RAM scanning has a bit of a change making it possible to use a side by side script such as we do.

**Comment by deezysleazy** [comment]
i got vaced, not really safe worked for a few games, i was hiding it, friends didn't even notice

**Comment by iPredictable** [comment]
@deezysleazy you cant get vac'd u mightve gotten overwatched or vac'd from something else

**Comment by deezysleazy** [comment]
can't u get overwatched from it?

**Comment by Moeinich** [comment]
@PredictEr I dont understand how this can stay undetected. Is mac really that different? Isn't VAC present at all on OSX and if it isn't, why wouldn't any of the larger cheat companies start making hacks for mac, surely its easy if it can't be detected..

**Comment by loco41211** [comment]
Are you guys sure about VAC not working on Mac? Has anyone dug into how VAC on Mac works? Maybe it does but has limitations that it doesn't have on windows.

**Comment by BlackCrab** [comment]
@Shadiezz @loco41211 Valve hasn't developed a proper VAC system for osx, and big companies aren't interested in mac osx hacks because not many coders know how to code mac hacks and 90% of the community uses windows.

**Comment by parhxm** [comment]
So is this not vaccable? I won't get vac banned for this?

**Comment by iPredictable** [comment]
@parhxm No as long as you dont mess with the files and dont run cheat engine or xCode while playing on VAC servers, Theres no way VAC can detect mac hacks.

**Comment by parhxm** [comment]
@PredictEr I'm trying to do it but it just doesn't work what do I do? It was working fine until yesterday.

**Comment by aKalisch** [comment]
This is simple to answer. VAC is on macOS. But it runs at Ring3 (User). The Hack must run as Root to access memory. So a User process wouldn't notice any root tasks.

On macOS you could only get overwatch or untrusted bans. So you just need to worry about Overwatch as there are just a few things to get untrusted bans (eg. writing unclamped angles to memory).

Long answer short: you can't get VAC banned on macOS. And if you are not completly stupid you can't get even overwatch banned (never got ow banned until using hack).

**Comment by aKalisch** [comment]
@PredictEr don't spread around false info. For xCode you would never get VAC banned.
And there is cheat engine for mac (but it's bad). But it also requires root (as bit slicer also did). So no VAC ban for this. I explained how you could get banned in CS:GO

**Comment by mptje88** [comment]
Ive been running this hack for like 2 months. made some few modifications to the script myself in xcode. No bans nothing only its crashes a shitload. Playing on Lem-supreme lvl :) no problems at all just besides the crashes.

**Comment by makikg123** [comment]
Hey, how can i change the toggle key?

**Comment by mptje88** [comment]
@makikg123

In the code itself

**Comment by Galenika** [comment]
@mptje88 
OH Thank you thought i changed it in cs go folder.
What modifications did you do to it?
CSGO GLOW HACK BY MPTJE88

You want me to clarify? 
in main.cpp this is outdated:
/**
 *
 * @toggle key : CTRL + ALT (OPT) + V
 * Change modifier at line 226
 * Change keycode at line 227
 *
 */

Solution:
If you want "V" only to toggle open keyboard.h use this only:

if (keycode == keyCodeForKeyString("v"))

**Comment by mptje88** [comment]
I guess so try it out, i only you gotta change it in the code. I dont use toggle so :p

**Comment by Galenika** [comment]
@mptje88 Play smart and start using toggle or one day you will get OW BAN.

**Comment by mptje88** [comment]
I play this game smart Af :) it just use it to know the enemy position. My original skill is lem anyway so i dont prefire. I o ly aim on them until i actually see them hehe.  It for map awareness this is neat

**Comment by CookinFood** [comment]
Wait so there is no way for me to get vac banned unless it is from overwatch right?

**Comment by mptje88** [comment]
@CookinFood 
Correct. There is no way for Valve to detect this hack. the beauty of how the mac works :)
Yes you can be banned from overwatch but its all up to you, how you use this hack :) So far 3 months playing with this hack on global now. No problems.

**Comment by CookinFood** [comment]
ok thank you

**Comment by ghost** [comment]
Vac did not detect it however I was banned via Overwatch with this hack after maybe half a year. (was a contributor until than)

**Comment by mptje88** [comment]
6 months in with this hack now. No bans nothing :)

**Comment by gabsens** [comment]
@mptje88  Valve doing a great job, as always

**Comment by iPredictable** [comment]
VAC VAC VAC VAC VAC VAC VAC VAC VAC

**Comment by fadedCS** [comment]
lmfao i got vacced http://steamcommunity.com/profiles/76561198332416829 and no, i only use a mac :3

**Comment by fadedCS** [comment]
but, it was game banned from 60 days ago and now the game ban is replaced with a recent vac ban (i hadn't been on the acc for 50 days, but i logged in to remove the number and saw i was recently vacced)

**Comment by amcgregor** [comment]
> There is no way for Valve to detect this hack. the beauty of how the mac works.

This is incorrect and misleading.  Sure, macOS is UNIX, but this does not mean a slight mistake by the user won't reveal what's running.  There is no particular reason why this would not be detectable.

It's a process spawned by your user (via sudo, sure), which means you'd need read access to it, and read access to wherever you put it.  Even if the process is running as root, and not directly accessible by other non-root processes, you can `lsof`, find it on disk, then hash/pattern match from there.  (Simply getting a list of all open files, then interrogating the files themselves, is infinitely easier than trying to scan other process memory.)

The source code, if compiled from it, would also generally be readable by your user unless precautions are taken.  If you can read it, software you run can read it, esp. as Steam is not a sandboxed App Store app.  Also: Spotlight will index it, making discovery as simple as asking the OS to do a search.  Now, this doesn't mean they actually _do_ go looking for things, but… really, it's only a matter of time.

It never hurts to be extra paranoid, e.g. using a bootable external drive to house the build environment you use to compile it, then making sure to move the resulting executable build artifact out of the default location it gets saved (buried in ~/Library)… and mark it owned by root and unreadable by ordinary users.  Practice good downloaded software hygiene.  ;)
